chainio/clients/fireblocks/client.go

```diff
@@ -0,0 +1,162 @@
+package fireblocks
+
+import (
+	"bytes"
+	"context"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/golang-jwt/jwt"
+	"github.com/google/uuid"
+)
+
+type FireblocksTxID string
+
+type FireblocksClient interface {
+	ContractCall(ctx context.Context, body *ContractCallRequest) (*ContractCallResponse, error)
+	ListContracts(ctx context.Context) ([]WhitelistedContract, error)
+}
+
+type fireblocksClient struct {
+	apiKey     string
+	privateKey *rsa.PrivateKey
+	baseURL    string
+	timeout    time.Duration
```

Comment:
timeout is used in both `http.Client{Timeout: timeout}` as well as the jwt token expiration. Feels like those two things are completely unrelated and should be different parameters? Unless I'm missing something..

Reply:
I see. Can't you request a JWT token for an hour or so and then use it for multiple calls? Or are you forced to request a new one on every call? Seems very inneficient.

Reply:
I think you pretty much have to generate a new token for every call since the token signs the path and body.